src/Stormpath/Authc/UsernamePasswordRequest.php

```diff
@@ -22,12 +22,15 @@ class UsernamePasswordRequest implements AuthenticationRequest
     private $username;
     private $password;
     private $host;
+    private $accountStore;
 
-    public function __construct($username, $password, $host = null)
+    public function __construct($username, $password, array $options = array())
     {
-        $this->host = $host;
         $this->password = $password ? str_split($password) : array();
         $this->username = $username;
+
+        $this->host = isset($options['host']) ? $options['host'] : null;
+        $this->accountStore = isset($options['accountStore']) ? $options['accountStore'] : null;
```

Comment:
We should probably assert that the accountStore is an instance of AccountStore here (might need to create an interface for this though).